### PR TITLE
Track all accounts seen in blocks/transactions/snarks

### PR DESCRIPTION
**Pull request by sosedoff**
https://www.notion.so/figmentnetworks/b47c0f8b935741d8ac28717ddeea2038?v=24513618c768490f8c7d63a6d5b156b3&p=16ad80ee89c04a3489a20bb4a014c895

### CODE REVIEW

File: indexing/data.go [hunk 1]
```diff
@@ -15,3 +15,31 @@ type Data struct {
 	Transactions    []model.Transaction
 	SnarkJobs       []model.SnarkJob
 }
+
+// AccountIDs returns a list of all accounts seen in the data payload
+func (d Data) AccountIDs() []string {
```
Comment (eesmerdag) [inline indexing/data.go:20]: Data struct has more fields which include some more account info. don't we need them?

Comment (sosedoff) [reply indexing/data.go:20]: Which fields are you referring to? We handle block producer, transactions and snarkers, dont think there's anything else that references accounts

Comment (eesmerdag) [reply indexing/data.go:20]: Prover is missing if I am not mistaken.

https://github.com/figment-networks/mina-indexer/blob/3d6b835ce05ed57a9c23a547952a90f4148531a0/indexing/data.go#L16

Comment (sosedoff) [reply indexing/data.go:20]: We already do that via `Snarkers` https://github.com/figment-networks/mina-indexer/blob/master/model/mapper/snarker.go#L42

Comment (eesmerdag) [reply indexing/data.go:20]: lgtm then sir

**Review by eesmerdag — APPROVED** [review]

**Pull request merged**